curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md

```diff
@@ -17,439 +17,542 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
+3 times "Hello".
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How many times will the function call itself?
+
+```javascript
+function mystery(n) {
+    if (n <= 1) return 1;
+    return mystery(n - 2);
+}
+mystery(5);
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+`2`
 
 ---
 
-Placeholder distractor 2
+`5`
 
 ---
 
-Placeholder distractor 3
+`4`
 
 #### --answer--
 
-Placeholder answer
+`3`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What feature do browsers implement to prevent infinite recursion?
 
 #### --distractors--
 
-Placeholder distractor 1
+Memory Cleanup.
 
 ---
 
-Placeholder distractor 2
+Automatic loop breaking.
 
 ---
 
-Placeholder distractor 3
+Thread termination.
 
 #### --answer--
 
-Placeholder answer
+Maximum call stack size.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is NOT true about recursion?
 
 #### --distractors--
 
-Placeholder distractor 1
+It must have a base case.
 
 ---
 
-Placeholder distractor 2
+It can replace loops.
 
 ---
 
-Placeholder distractor 3
+It can be memory-intensive.
 
 #### --answer--
 
-Placeholder answer
+It always returns 0.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In recursive event handling, what's the risk of?
+
+```javascript
+element.onclick = function() {
+    this.onclick();
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+Infinite event loop.
 
 ---
 
-Placeholder distractor 2
+Browser crash.
 
 ---
 
-Placeholder distractor 3
+Memory leak.
 
 #### --answer--
 
-Placeholder answer
+All of the above.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What's wrong with this recursive DOM traversal?
+
+```javascript
+function findElement(element) {
+    if (element.id === 'target') return element;
+    findElement(element.firstChild);
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+Missing base case.
 
 ---
 
-Placeholder distractor 2
+Wrong parameter.
 
 ---
 
-Placeholder distractor 3
+Incorrect comparison.
 
 #### --answer--
 
-Placeholder answer
+Missing return for recursive call.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+When using recursion, which statement is true?
 
 #### --distractors--
 
-Placeholder distractor 1
+The function always runs infinitely.
 
 ---
 
-Placeholder distractor 2
+The function uses a while loop.
 
 ---
 
-Placeholder distractor 3
+The function has no parameters.
 
 #### --answer--
 
-Placeholder answer
+The function calls itself.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What will this return?
+
+```javascript
+function join(arr) {
+    if (arr.length <= 1) return arr;
+    return [arr[0]].concat(join(arr.slice(1)));
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+A reversed array.
 
 ---
 
-Placeholder distractor 2
+The first element only.
 
 ---
 
-Placeholder distractor 3
+An empty array.
 
 #### --answer--
 
-Placeholder answer
+A copy of the original array.
```

Comment:
Unless original array has only single element or is empty, in which case it will be the same original array.